src/dbnode/storage/index.go

```diff
@@ -1129,22 +1173,35 @@ func (i *nsIndex) query(
 	state.Unlock()
 
 	if err != nil {
+		sp.LogFields(opentracinglog.Error(err))
 		return false, err
 	}
 
 	return exhaustive, nil
 }
 
 func (i *nsIndex) execBlockQueryFn(
+	ctx context.Context,
 	cancellable *resource.CancellableLifetime,
 	block index.Block,
 	query index.Query,
 	opts index.QueryOptions,
 	state *asyncQueryExecState,
 	results index.BaseResults,
 ) {
-	blockExhaustive, err := block.Query(cancellable, query, opts, results)
+	ctx, sp := ctx.StartTraceSpan(tracepoint.NSIdxBlockQuery)
+	sp.LogFields(
+		opentracinglog.String("query", query.String()),
+		opentracinglog.String("namespace", i.nsMetadata.ID().String()),
+		opentracinglog.Int("limit", opts.Limit),
+		xopentracing.Time("start", opts.StartInclusive),
+		xopentracing.Time("end", opts.EndExclusive),
+	)
+	defer sp.Finish()
+
+	blockExhaustive, err := block.Query(ctx, cancellable, query, opts, results)
 	if err == index.ErrUnableToQueryBlockClosed {
+		sp.LogFields(opentracinglog.Error(err))
```

Comment:
I wouldn't log this error, this is an expected error (as explained by the comment below).

Reply:
Yeah was gonna ask about this error and a couple others. Sounds good.